cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/GDPRConsentLib.java

```diff
@@ -344,14 +329,15 @@ private void renderMsgAndSaveConsent() throws ConsentLibException {
             @Override
             public void onSuccess(Object result) {
                 try{
-                    JSONObject jsonResult = (JSONObject) result;
+                    JSONObject jsonResult = new JSONObject((String) result);
                     consentUUID = jsonResult.getString("uuid");
                     metaData = jsonResult.getString("meta");
                     if(jsonResult.has("msgJSON") && !jsonResult.isNull("msgJSON")) {
                         setNativeMessageView(jsonResult.getJSONObject("msgJSON"));
                         showView(nativeView);
                     } else if(jsonResult.has("url") && !jsonResult.isNull("url")){
-                        loadConsentUI(jsonResult.getString("url"));
+                        String url = jsonResult.getString("url");
+                        loadConsentUI(url);
```

Comment:
did you perhaps add the intermediate `url` variable for debugging and forgot inline back inside `loadConsentUI`?

Reply:
I was actually testing your attention there =P
Gonna change it 😅